packages/eslint-plugin-react-hooks/src/RulesOfHooks.ts

```diff
@@ -187,15 +185,18 @@ export default {
       onCodePathSegmentEnd: () => codePathSegmentStack.pop(),
 
       // Maintain code path stack as we traverse.
-      onCodePathStart: () => codePathReactHooksMapStack.push(new Map()),
+      onCodePathStart: () =>
+        codePathReactHooksMapStack.push(
+          new Map<Rule.CodePathSegment, Node[]>(),
+        ),
 
       // Process our code path.
       //
       // Everything is ok if all React Hooks are both reachable from the initial
       // segment and reachable from every final segment.
       onCodePathEnd(codePath, codePathNode) {
         const reactHooksMap = codePathReactHooksMapStack.pop();
-        if (reactHooksMap.size === 0) {
+        if (!reactHooksMap?.size) {
```

Comment:
Could we keep the previous comparison to 0?
```suggestion
        if (reactHooksMap?.size === 0) {
```

Reply:
Since that would allow `reactHooksMap` to be `undefined` in the remainder of that function, should I do similar to what you recommended in the other comment, and throw if it's `undefined` here?

Reply:
Updated these to throw if unexpectedly undefined / null